### PR TITLE
Add retrofit docs

### LINKED ISSUE #88: Add retrofit integration docs and sample
We need to add the Helios retrofit integration to the microsite documentation and also add a little example on the `samples` module (or in a new module under the same folder)

### PR DESCRIPTION
**Pull request by jereksel**
Fixes #88 

This MR also:
- Removes null returning on error (replaced by exception throwing)
- Makes DecodingError extend Exception for much easier debugging
- Since `Class` implements `Type` retrieving name is not required

### CODE REVIEW

**Comment by AdrianRaFo** [comment]
Please remember to assign yourself the issue or comment in it if you are going to take it

**Comment by AdrianRaFo** [comment]
Pending on https://github.com/47deg/helios/pull/98

**Comment by AdrianRaFo** [comment]
@jereksel You should be able to uncomment the example now the issue has been fixed

File: helios-core/src/main/kotlin/helios/core/errors.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package helios.core
 
-sealed class DecodingError
+sealed class DecodingError: Exception()
```
[the remark this answers is not in the document]
Comment (jereksel) [reply helios-core/src/main/kotlin/helios/core/errors.kt]: I know, but current error handling is not recursive and it's difficult to debug errors in nested objects. That way I have a stacktrace with path to field that caused this error.

[the remark this answers is not in the document]
Comment (nomisRev) [reply helios-core/src/main/kotlin/helios/core/errors.kt]: Can you be more specific? And/or provide an example?
Introducing `Exception` here makes this impure which makes the whole library impure which contradictory to our goal.

Additionally this is caused by Retrofit and not Helios, so making a change to Helios to solve a Retrofit issue is something we shouldn't do.

**Comment by AdrianRaFo** [comment]
Superseded by https://github.com/47deg/helios/pull/105

File: helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt [hunk 2]
```diff
@@ -11,14 +10,10 @@ import java.nio.ByteBuffer
...
+  override fun convert(value: ResponseBody): T? = value.use { value ->
     Json
       .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
       .flatMap { it.decode(decoder) }
-      .getOrElse { null }
-  } catch (ex: Exception) {
-    null
-  } finally {
-    value.close()
+      .fold({ throw it }, { it })
```
[the remark this answers is not in the document]
Comment (AdrianRaFo) [reply helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt]: Addressed here https://github.com/47deg/helios/pull/105

**Pull request closed without merging**